File: solutions/53_Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        # Initialize
        result = nums[0]
        sum = 0

        for num in nums:

            # If previous sum < 0, we dont want to add it to current subarray
            # but restart a new subarray from current number
            if sum > 0:
                sum += num
            else:
                sum = num

            # Update result
            result = max(result, sum)
        return result

File: solutions/53_Maximum_Subarray.c
int maxSubArray(int* nums, int numsSize) {
    int results = nums[0], sum = 0;

    for (int i = 0; i < numsSize; i++) {
        if (sum < 0) {
            sum = nums[i];
        } else {
            sum += nums[i];
        }
        results = results < sum ? sum : results;
    }
    return results;
}